File: javascript/hamming/hamming.js

var Hamming = function() {
  this.compute = function(a, b) {
    if(a.length !== b.length) throw new Error('DNA strands must be of equal length.');
    var hDist = 0;
    for (var i = 0; i < a.length; i++) {
      if(a[i] !== b[i]) hDist++;
    }
    return hDist;
  };
};

module.exports = Hamming;

File: ruby/hamming/hamming.rb
module Hamming
  def self.compute(a, b)
    if a.length != b.length
      raise ArgumentError
    else
      
      dist = 0
      i = 0
      while (i < a.length)
        if a[i] != b[i]
          dist += 1
        end
        i += 1
      end

    end
    dist
  end
end

module BookKeeping
  VERSION = 3
end
